PowerPointLabs/PowerPointLabs/ColorPicker/MagnifierForm.cs

```diff
@@ -0,0 +1,212 @@
+
+            public MagnifierOverlay(int width, int height)
+            {
+                TopMost = true;
+                ShowInTaskbar = false;
+                BackColor = Color.LimeGreen;
+                TransparencyKey = BackColor;
+                FormBorderStyle = FormBorderStyle.None;
+                StartPosition = FormStartPosition.Manual;
+                MinimumSize = new Size(1, 1);
+
+                int outlineHalfSize = OVERLAY_OUTLINE_SIZE / 2;
+                Width = width + OVERLAY_OUTLINE_SIZE;
+                Height = height + OVERLAY_OUTLINE_SIZE;
+                positionOffset = new Point(-outlineHalfSize, -outlineHalfSize);
+
+                outlinePen = new Pen(Color.Black, OVERLAY_OUTLINE_SIZE);
+                outlinePath = new GraphicsPath();
+                outlinePath.AddEllipse(outlineHalfSize, outlineHalfSize, width, height);
+            }
+
+            public Point PositionOffset
+            {
+                get { return positionOffset; }
+            }
+
+            protected override void OnPaint(PaintEventArgs e)
+            {
+                e.Graphics.DrawPath(outlinePen, outlinePath);
+            }
+        }
+        #endregion
+
+        #region Public API
+        public new void Show()
+        {
+            if (isMagInitialized)
+            {
+                UpdateMagnifier();
+                timer.Start();
+                overlay.Show();
+                Visible = true;
+            }
+        }
+
+        public new void Hide()
+        {
+            timer.Stop();
+            overlay.Hide();
+            Visible = false;
+        }
+        #endregion
+
+        #region Handled events
+        private void Overlay_Shown(object sender, EventArgs e)
+        {
+            // Overlay can only be filtered after it is shown
+            AddMagnifierFilteredWindow(overlay.Handle);
+        }
+        
+        private void MagnifierForm_FormClosing(object sender, FormClosingEventArgs e)
+        {
+            TeardownMagnifier();
+        }
+
+        private void Timer_Tick(object sender, EventArgs e)
+        {
+            UpdateMagnifier();
+        }
+        #endregion
+
+        #region Magnifier methods
+        private void SetupMagnifier()
+        {
+            magFilteredWindows = new List<IntPtr>();
+            IntPtr hInst = Native.GetModuleHandle(null);
+
+            // Create Magnifier window
+            hwndMag = Native.CreateWindowEx(
+                (int)Native.ExtendedWindowStyles.WS_EX_LEFT, 
+                Native.WC_MAGNIFIER, "MagnifierWindow", 
+                (int)Native.WindowStyles.WS_CHILD |
+                (int)Native.WindowStyles.WS_VISIBLE |
+                (int)Native.MagnifierStyle.MS_CLIPAROUNDCURSOR,
+                0, 0, Width, Height, 
+                Handle, IntPtr.Zero, hInst, IntPtr.Zero);
+
+            if (hwndMag == IntPtr.Zero)
+            {
+                throw new Exception();
```

Comment:
and try to use the correct exception that describes the problem instead of the generic one